shared/ui/combobox/combobox.tsx

```diff
@@ -0,0 +1,161 @@
+
+import { Typography } from '..';
+
+import ArrowDown from '../../../assets/icons/arrow-down.svg?url';
+import Close from '../../../assets/icons/close.svg?url';
+
+import Image from 'next/image';
+
+type Option = {
+  label: string;
+  value: number | string;
+};
+
+export type ComboboxProps = {
+  disabled?: boolean;
+  errorMessage?: string;
+  inputValue: string;
+  label?: string;
+  name?: string;
+  onChange: (value: null | string) => void;
+  onClear?: () => void;
+  onInputChange: (value: string) => void;
+  options: Option[];
+  placeholder?: string;
+  showClearButton?: boolean;
+  value: string;
+};
+
+export const Combobox: FC<ComboboxProps> = props => {
+  const {
+    disabled,
+    errorMessage,
+    inputValue,
+    label,
+    name,
+    onChange,
+    onClear,
+    onInputChange,
+    options,
+    placeholder,
+    showClearButton = true,
+    value,
+  } = props;
+
+  const showError = !!errorMessage && errorMessage.length > 0;
+
+  const inputChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.currentTarget.value === '') {
+      onChange(null);
+    }
+    onInputChange(e.currentTarget.value);
+  };
+
+  const handleClearButtonClicked: MouseEventHandler<HTMLDivElement> = () => {
+    onInputChange('');
+    onChange(null);
+  };
+
+  const filteredOptions =
+    inputValue === ''
+      ? options
+      : options.filter(option => option.label.toLowerCase().includes(inputValue.toLowerCase()));
+
+  const getDisplayingValue = (value: number | string) =>
+    options?.find(option => option.value === value)?.label || '';
+
+  const s = {
+    container: style.container,
+    box: style.boxContainer,
+    input: clsx(
+      style.inputField,
+      style.input,
+      showError && style.error,
+      showClearButton && style.hasClearButton
+    ),
+    button: style.button,
+    clearButton: style.clearButton,
+    options: clsx(style.content, filteredOptions.length === 0 && style.empty),
+    item: style.item,
+    viewport: style.viewport,
+  };
+
+  return (
+    <>
+      <ComboboxUI
+        {...{
+          disabled,
+          name,
+          onChange,
+          value,
+        }}
+        as={'div'}
+        className={s.container}
+      >
+        <div className={s.box}>
+          {label && (
+            <Typography.Regular14 color={'var(--color-light-900)'}>{label}</Typography.Regular14>
+          )}
+
+          <ComboboxUI.Button as={'div'}>
+            <ComboboxUI.Input
+              className={s.input}
+              displayValue={getDisplayingValue}
+              onChange={inputChangeHandler}
+              placeholder={placeholder}
+            />
+
+            <div className={s.button}>
+              <Image alt={'arrow down'} src={ArrowDown} className={style.icon} />
+            </div>
+          </ComboboxUI.Button>
+
+          {showClearButton && !!value && (
+            <div className={s.clearButton} onClick={onClear ?? handleClearButtonClicked}>
```

Comment:
The `onClear` callback is not used in the component code. Its presence should be reconsidered